### DISCUSSION TITLE
Multichannel option doesn't seem to work as expected

### DISCUSSION THREAD

**Discussion by KenDFish** [deepgram/community#527] [General help]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

We're using the .NET API to transcript pre-recorded audio from a call centre. The files are typically two channel OPUS files: customer on one channel, agent on the other (and, very occasionally, a 3rd agent on a 3rd channel). We'd like seperated transcripts of both sides: so a transcript of the customer, and a transcript of the agent.

This seems to match the example given on the [https://developers.deepgram.com/docs/multichannel-vs-diarization](url) page. The returned result doesn't quite match the example format given: it has two channels, but both channels contain effectively the same transcript: both channels, rather than just the transcript for that one channel.  The "hierarchy" of the returned JSON is: 

![image](https://github.com/deepgram/community/assets/6280435/7c6af8ca-4a7b-4e16-8d18-a8675149401e)

Turning on Diarization helps most of the time, but from the above link this shouldn't be necessary - and it doesn't always work.

What do we need to do to ensure the returned result contains distinct transcripts for each channel?

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen?model=nova-2-phonecall&multichannel=true&numerals=true&dates=true&smart_format=true&paragraphs=true

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

2768db3d-5f90-46e5-b167-9bb9c6b0d06b

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jjmaldonis** [comment]
Hey @KenDFish, the audio is identical on both channels so that's why you're getting the same transcript on both channels.

**Reply by KenDFish** [reply]
Apologies - you're right. We have two mono OPUS files as a source, which are merged into a stereo OPUS file before uploading: it seems that ffmpeg has a bug that means that the two files aren't mixed complete to the "left" and "right" channels - so the audio from the other channel leaks through. Changing the output to MP4 seems to fix this issue - no big deal to us, and Deepgram now returns separate transcripts per channel as expected.

The only issue now is that the paragraph detection doesn't appear to work as we'd expect.  Regardless of the Diarization settings, the paragraphs aren't split when there's audio on the other channel.

An example request is e4a6cd95-117a-4f51-8036-a4f204f0e2e3 - in this example, there's audio on the other channel after sentence 1, so I'd expect that to trigger a new paragraph:

![image](https://github.com/deepgram/community/assets/6280435/f28f3733-e6e0-4d36-97fe-cc094e8af175)

Is there something we can do to improve the paragraph splitting (which seemed to be working better with the OPUS files)

**Reply by jjmaldonis** [reply]
We don't split paragraphs when another person speaks on a different channel because, when two people speak at the same time, you'd get a bunch of single-word paragraphs. Instead, we process the paragraphs for each channel independently. It's possible the speakers in the non-opus files took turns speaking, which will result in very accurate paragraphs.

If you'd like to split the paragraphs when the speaker changes, you can add custom post processing logic to the transcripts. Just be aware of the edge case I mentioned where two people are speaking at the same time :)

**Reply by KenDFish** [reply]
I think I’ve accidentally confused us all with the OPUS problem. I now realise that the OPUS files we’re producing had the “cross channel” problem: audio from one channel was also present (although fainter) in the other channel, due to faulty stereo mixing. As you’ve said, this meant we were effectively getting two transcripts of two near-identical channels, and it was the Diarization feature that was working its magic and doing the speaker detection, rather than any channel attribution. Apologies again for the confusion!

However, Diarization did seem to do a better job of splitting paragraphs. From [https://developers.deepgram.com/docs/paragraphs](url) ..

> When the Multichannel feature is enabled, paragraphs breaks are influenced by channel changes.

So, with that in mind, here’s the start of the file we’re testing with, which I've annotated with the words actually being spoken:

![image](https://github.com/deepgram/community/assets/6280435/8d57e43f-01a4-4d43-9f33-15bad7c5183a)

1. Channel 0 speaks at 1.8s until 3.3s
2. Channel 1 speaks at 3.8s until 4.8s
3. Channel 0 speaks at 5.3s

… so what we’re expecting is 3 paragraphs: 

1. first paragraph is channel 0
2. second is channel 1
3. third is back to channel 0.

However, in the output, there’s no break when channel 1 speaks - channel 0's audio is one paragraph:
```

"paragraphs": [
                {
                  "sentences": [
                    {
                      "text": "Hi.",
                      "start": 1.76,
                      "end": 1.92
                    },
                    {
                      "text": "Can I speak to Tim, please?",
                      "start": 1.92,
                      "end": 3.46
                    },
                    {
                      "text": "Good morning.",
                      "start": 5.04,
                      "end": 5.6
                    },
                    {
                      "text": "It's Charlotte calling from [redacted]",
                      "start": 5.6,
                      "end": 7.4520974
                    },

```

As you can see from the audio, the two channels aren’t talking over each other, and as the documentation suggests “paragraphs breaks are influenced by channel changes”, this looks like a case where this didn’t happen. Am I misunderstanding the documentation? We can do our own paragraph processing, but there’s no point in reinventing the wheel if there’s some small change we can make to fix this within DeepGram.

I understand the "speaking over each other" problem: when we were testing this with the (faulty!) Diarized OPUS files, the paragraph split seemed to do a great job: when two parties talked over each other, the paragraphs were still split cleanly: so if person 1 talked over person 2, we saw a complete paragraph from person 1, and another complete paragraph from person 2 - just the start offsets indicated that the paragraphs had overlapped.
